web-common/src/components/editor/presets/yamlWithJsonAndSql.ts

```diff
@@ -0,0 +1,38 @@
+import { jsonLanguage } from "@codemirror/lang-json";
+import { SQLite } from "@codemirror/lang-sql";
+import { yamlLanguage } from "@codemirror/lang-yaml";
+import { LRLanguage } from "@codemirror/language";
+import { parseMixed } from "@lezer/common";
+
+const activateOnNodes = new Set(["BlockLiteralContent"]);
+
+let blockLiteralCount = 0;
+
+const wrap = parseMixed((node) => {
+  if (node?.name === "Stream") {
+    blockLiteralCount = 0;
+  }
+  if (!node || !activateOnNodes.has(node.name)) {
+    return null;
+  }
+  blockLiteralCount++;
+
+  if (blockLiteralCount === 1) {
+    return {
+      parser: SQLite.language.parser,
```

Comment:
We've defined a CodeMirror SQL dialect for DuckDB in `web-common/src/features/models/workspace/Editor.svelte`. You could move it to a common location then import it here?